app/models/event_schedule.rb

```diff
@@ -52,4 +55,34 @@ def start_before_start_hour
   def conference_id
     schedule.program.conference_id
   end
+
+  ##
+  # Validates that the event is scheduled in the same room as its track
+  #
+  def same_room_as_track
+    if event && event.track.try(:room) && event.track.room != room
```

Comment:
You could also try this approach to make it easier to understand what your code is doing. This way when reading the line that adds the error you are focusing on the actual validation that matters (in this case the room being the same as the track room)
```
return false unless event && event.track && event.track.room
errors.add() unless event.track.room == room
```